dango/taxman/src/execute.rs

```diff
@@ -57,19 +59,39 @@ fn pay(ctx: MutableCtx, ty: FeeType, payments: BTreeMap<Addr, Coins>) -> anyhow:
         total
     );
 
-    // For now, nothing to do.
-    // In the future, we will implement affiliate fees.
-    let events = payments
-        .into_iter()
-        .map(|(user, amount)| {
-            ContractEvent::new(&ReceiveFee {
-                handler: ctx.sender,
+    let oracle = ctx.querier.query_dango_config()?.addresses.oracle;
+
+    // Record the fees in storage and emit events.
+    let mut events: Vec<ContractEvent> = Vec::new();
```

Comment:
@apollo-sturdy we have `grug::EventBuilder` which may be better suited for this.